drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,668 @@
+        return -EINVAL;
+
+    sx126x_set_tx_params(dev, pow, SX126X_RAMP_10_US);
+    return 0;
+}
+
+static int _off(ieee802154_dev_t *hal)
+{
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    sx126x_set_sleep(dev, SX126X_SLEEP_CFG_COLD_START);
+    return 0;
+}
+
+static int _config_addr_filter(ieee802154_dev_t *hal, ieee802154_af_cmd_t cmd, const void *value)
+{
+    (void)cmd;
+    (void)value;
+    uint16_t pan_id = *(uint16_t*)value;
+    sx126x_t *dev = hal->priv;
+    switch(cmd) {
+        case IEEE802154_AF_SHORT_ADDR:
+            memcpy(dev->short_addr, value, IEEE802154_SHORT_ADDRESS_LEN);
+            break;
+        case IEEE802154_AF_EXT_ADDR:
+            memcpy(dev->long_addr, value, IEEE802154_LONG_ADDRESS_LEN);
+            break;
+        case IEEE802154_AF_PANID:
+            dev->pan_id = pan_id;
+            break;
+        case IEEE802154_AF_PAN_COORD:
+            return -ENOTSUP;
+    }
+    return 0;
+}
+
+static int _request_on(ieee802154_dev_t *hal)
+{
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    _set_state(dev, STATE_IDLE);
+    return 0;
+}
+
+static int _confirm_on(ieee802154_dev_t *hal)
+{
+    (void)hal;
+
+    return -ENOTSUP;
+}
+
+static int _set_cca_mode(ieee802154_dev_t *hal, ieee802154_cca_mode_t mode)
+{
+    (void)hal;
+    sx126x_t* dev = hal->priv;
+    DEBUG("[sx126x] netdev: set_cca_mode \n");
+        dev->cad_params.cad_exit_mode = SX126X_CAD_ONLY,
+        dev->cad_params.cad_detect_min = 10,
+        dev->cad_params.cad_detect_peak = 22,
+        dev->cad_params.cad_symb_nb = SX126X_CAD_02_SYMB,
+        dev->cad_params.cad_timeout = 0x000F00;
+    (void)mode;
+
+    return 0;
+}
+
+static int _config_src_addr_match(ieee802154_dev_t *hal, ieee802154_src_match_t cmd, const void *value)
+{
+    (void)hal;
+    (void)cmd;
+    (void)value;
+
+    return -ENOTSUP;
+}
+
+static int _set_frame_filter_mode(ieee802154_dev_t *hal, ieee802154_filter_mode_t mode)
+{
+    sx126x_t* dev = hal->priv;
+
+    bool ackf = true;
+    bool _promisc = false;
+
+    switch (mode) {
+        case IEEE802154_FILTER_ACCEPT:
+            DEBUG("Filter_accept_all\n");
+            break;
+        case IEEE802154_FILTER_PROMISC:
+            _promisc = true;
+            break;
+        case IEEE802154_FILTER_ACK_ONLY:
+            ackf = false;
+            DEBUG("Filter_ack_only\n");
+            break;
+        default:
+            return -ENOTSUP;
+    }
+
+    dev->ack_filter = ackf;
+    dev->promisc = _promisc;
+
+    return 0;
+}
+
+static int _set_csma_params(ieee802154_dev_t *hal, const ieee802154_csma_be_t *bd, int8_t retries)
+{
+    (void)hal;
+    (void)bd;
+    (void)retries;
+
+    return -ENOTSUP;
+}
+
+
+static const ieee802154_radio_ops_t sx126x_ops = {
+    .caps =  IEEE802154_CAP_SUB_GHZ
+          | IEEE802154_CAP_IRQ_CRC_ERROR
+          | IEEE802154_CAP_IRQ_RX_START
+          | IEEE802154_CAP_IRQ_TX_DONE
+          | IEEE802154_CAP_IRQ_CCA_DONE
+          | IEEE802154_CAP_PHY_BPSK,
```

Comment:
This driver exposes only PHY. We should consider adding a `IEEE802154_CAP_PHY_LORA` for this.